### PR TITLE
add support for rke2 on x86_64 architecture

### PR DESCRIPTION
**Pull request by alknopfler**
Changes:
- New structure cluster agnostic in env file
- Readme updated
- rke2 feature available for single node and ha with kube-vip

### CODE REVIEW

File: slemicro/combustion/script [hunk 1]
```diff
@@ -448,7 +499,7 @@ if [ "${UPDATEANDREBOOT}" = true ]; then
 	Description=Reboot if required once
 	Wants=network-online.target
 	# If services doesn't exist is ok
-	After=network.target network-online.target k3s_installer.service rancher_installer.service skip-rancher-bootstrap.service elemental_installer.service
+	After=network.target network-online.target rancher_installer.service skip-rancher-bootstrap.service elemental_installer.service
```
Comment (e-minguez) [inline slemicro/combustion/script:502]: Same as before, make it dependant of k3s or rke2 depending on which one has been installed.

File: slemicro/create_vm.sh [hunk 2]
```diff
@@ -51,6 +52,16 @@ if [ "${REGISTER}" == false ] && [ "${RANCHERBOOTSTRAPSKIP}" == true ]; then
 	die "RANCHERBOOTSTRAPSKIP requires REGISTER" 3
 fi
 
+# Check if cluster installation to set correct path to kubeconfig and kubectl
+if ${K3S} ; then
+  		export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+  		export KUBECTL=/usr/local/bin/kubectl
+else # RKE2
```
Comment (e-minguez) [inline slemicro/create_vm.sh:59]: I'd say to do a switch instead and fail if this is not rke2 or k3s or false

File: slemicro/env.example [hunk 3]
```diff
@@ -21,6 +21,17 @@ export INSTALL_K3S_EXEC="server --cluster-init --write-kubeconfig-mode=644"
 # To add worker nodes:
 # export INSTALL_K3S_EXEC="agent --server https://${VIP}:6443"
 export K3S_TOKEN="foobar"
+export RKE2=true
```
Comment (e-minguez) [inline slemicro/env.example:24]: RKE2 and K3S cannot be enabled at the same time

**Review by e-minguez — CHANGES_REQUESTED** [review]

File: slemicro/combustion/script [hunk 4]
```diff
@@ -272,18 +323,18 @@ if [ "${RANCHERFLAVOR}" != false ]; then
...
 	# Create a systemd unit to install rancher once
-	# Using "User=root" is required for some environment variables to be present 
+	# Using "User=root" is required for some environment variables to be present
 	cat <<- EOF > /etc/systemd/system/rancher_installer.service
 	[Unit]
-	Description=Deploy Rancher on K3s
+	Description=Deploy Rancher on K3s/RKE2
 	Wants=network-online.target
-	After=network.target network-online.target k3s.service
+	After=network.target network-online.target
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply slemicro/combustion/script:331]: I think with the ${KUBECTL} wait is far enough, assuming than the timeout is not set

**Comment by alknopfler** [comment]
Tests realized:
- Cluster envvar false -> Only micro installation: OK
- Cluster envvar not valid: OK
image

- Micro + k3s without kube-vip (single node) + rancher + elemental: OK
image

File: slemicro/create_vm.sh [hunk 5]
```diff
@@ -156,6 +176,7 @@ else
 	die "VM not deployed. Unsupported operating system" 2
 fi
 
+
```
Comment (e-minguez) [inline slemicro/create_vm.sh:179]: Extra line not needed :P

File: slemicro/combustion/script [hunk 6]
```diff
@@ -183,17 +228,23 @@ if [ "${KUBEVIP}" = true ]; then
...
+	          value: "20"
 	        - name: vip_retryperiod
-	          value: "1"
+	          value: "4"
 	        - name: address
 	          value: ${VIP}
-	        - name: prometheus_server
-	          value: :2112
+	EOF
+if [ ${CLUSTER} == "k3s" ]; then
```
Comment (e-minguez) [inline slemicro/combustion/script:239]: This needs to be tabbed I guess?
IDK if there is another way to do this, but it feels a little bit _dirty_ 😓 let me think about this.

File: slemicro/README.md [hunk 7]
```diff
@@ -72,20 +73,20 @@ export SSHPUB="${HOME}/.ssh/id_rsa.pub"
...
-# export INSTALL_K3S_EXEC="agent --server https://${VIP}:6443"
-# K3s token
-export K3S_TOKEN="foobar"
+# Set it to false if you don't want K3s or rke2 to be deployed ["k3s"|"rke2"|false]
+export CLUSTER="k3s"
+# Specify a k3s or rke2 cluster version ["v1.25.8+k3s1"|"v1.25.9+rke2r1"]
+export INSTALL_CLUSTER_VERSION="v1.25.8+k3s1"
+# For the first node in an HA environment. This INSTALL_CLUSTER_EXEC will be translated to INSTALL_K3S_EXEC or INSTALL_CLUSTER_EXEC depending on the value of CLUSTER
+# export INSTALL_CLUSTER_EXEC="server --cluster-init --write-kubeconfig-mode=644 --tls-san=${VIP} --tls-san=${VIP}.sslip.io"
+# For a single node. This INSTALL_CLUSTER_EXEC will be translated to INSTALL_K3S_EXEC or INSTALL_CLUSTER_EXEC depending on the value of CLUSTER
```
Comment (e-minguez) [inline slemicro/README.md:82]:
```suggestion
# For a single node. This INSTALL_CLUSTER_EXEC will be translated to INSTALL_K3S_EXEC or INSTALL_RKE2_EXEC depending on the value of CLUSTER
```

File: slemicro/README.md [hunk 8]
```diff
@@ -72,20 +73,20 @@ export SSHPUB="${HOME}/.ssh/id_rsa.pub"
...
+export INSTALL_CLUSTER_EXEC="server --cluster-init --write-kubeconfig-mode=644"
+# To add control plane nodes. This INSTALL_CLUSTER_EXEC will be translated to INSTALL_K3S_EXEC or INSTALL_CLUSTER_EXEC depending on the value of CLUSTER:
```
Comment (e-minguez) [inline slemicro/README.md:84]:
```suggestion
# To add control plane nodes. This INSTALL_CLUSTER_EXEC will be translated to INSTALL_K3S_EXEC or INSTALL_RKE2_EXEC depending on the value of CLUSTER:
```

File: slemicro/combustion/script [hunk 9]
```diff
@@ -225,13 +277,11 @@ if [ "${RANCHERFLAVOR}" != false ]; then
 	cat <<- EOF > /usr/local/bin/rancher_installer.sh
 	#!/bin/bash
 	set -euo pipefail
-	
-	# Wait for k3s to be available
-	until [ -f /etc/rancher/k3s/k3s.yaml ]; do sleep 2; done
-	export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+	# Wait for cluster to be available
+	until [ -f ${KUBECONFIG} ]; do sleep 2; done
+	export KUBECONFIG=${KUBECONFIG}
```
Comment (e-minguez) [inline slemicro/combustion/script:282]: Again, this is not needed :D

**Review by e-minguez — CHANGES_REQUESTED** [review]

**Pull request merged**